src/spurt/io/_three_d.py

```diff
@@ -0,0 +1,263 @@
+"""Handle 3D interfaces for unwrapping."""
+
+from __future__ import annotations
+
+import numpy as np
+from numpy.typing import ArrayLike, DTypeLike
+
+from ._interface import InputInterface, InputStackInterface
+
+__all__ = [
+    "Reg3DInput",
+    "Irreg3DInput",
+]
+
+
+class Reg3DInput(InputStackInterface):
+    """A single numpy 3D array as input.
+
+    A 2D image will just be a 3D array with size of time axis set to 1. This is
+    largely a helper class to assist in development. `time_dim` is an optional
+    input to indicate time axis.
+    """
+
+    def __init__(
+        self,
+        data: InputInterface,
+        time_dim: int = 0,
+    ):
+        """Initialize a stack interface for a numpy array.
+
+        Representing 3D data in a cube - regularly spaced in spatial dimension.
+
+
+        Parameters
+        ----------
+        data: np.ndarray
+            Numpy 3D array as input
+
+        time_dim: int, optional
+            Axis of the numpy array that represents the time dimension
+        """
+        # Check number of dimensions of input array
+        if data.ndim != 3:
+            clsname = type(self).__name__
+            errmsg = (
+                f"{clsname} only supports 3D arrays. Unsupported shape {data.shape}"
+            )
+            raise ValueError(errmsg)
+
+        # Check if time_dim is within limits
+        if time_dim >= data.ndim:
+            clsname = type(self).__name__
+            errmsg = f"{clsname} received time dimension outside array dimensions"
+            raise ValueError(errmsg)
+
+        # We only hold a reference to the data
+        self._data = data
+        self._time_dim = time_dim
+
+    @property
+    def dtype(self) -> DTypeLike:
+        return self._data.dtype
+
+    @property
+    def ndim(self) -> int:
+        return 2
+
+    @property
+    def time_dim(self) -> int:
+        return int(self._time_dim > 0)
```

Comment:
Is there any reason that this couldn't just be `return self._time_dim`?

Reply:
If we pass in a `3D array` and `time_dim` could be 1 or 2. 2 is the more common use case - we flatten the spatial dimension for regular grid.